readthedocs/rtd_tests/tests/test_build_forms.py

```diff
@@ -147,3 +157,182 @@ def test_resources_are_deleted_when_version_is_inactive(
         remove_search_indexes.delay.assert_called_once()
         self.assertFalse(version.imported_files.exists())
         self.assertTrue(another_version.imported_files.exists())
+
+    def test_change_slug(self):
+        version = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="slug",
+        )
+
+        test_slugs = [
+            "anotherslug",
+            "another_slug",
+            "another-slug",
+        ]
+        for slug in test_slugs:
+            form = VersionForm(
+                {
+                    "slug": slug,
+                    "active": True,
+                    "privacy_level": PUBLIC,
+                },
+                instance=version,
+                project=self.project,
+            )
+            assert form.is_valid()
+            assert version.slug == slug
+
+    def test_change_slug_wrong_value(self):
+        version = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="slug",
+        )
+
+        test_slugs = (
+            "",
+            "???//",
+            "long" * 100,
+            "Slug-with-uppercase",
+            "no-ascíí",
+            "with spaces",
+            "-almos-valid",
+            "no/valid",
+        )
+        for slug in test_slugs:
+            form = VersionForm(
+                {
+                    "slug": slug,
+                    "active": True,
+                    "privacy_level": PUBLIC,
+                },
+                instance=version,
+                project=self.project,
+            )
+            assert not form.is_valid()
+            assert "slug" in form.errors
+
+    def test_change_slug_already_in_use(self):
+        version_one = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="one",
+        )
+        version_two = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="two",
+        )
+
+        form = VersionForm(
+            {
+                "slug": version_two.slug,
+                "active": True,
+                "privacy_level": PUBLIC,
+            },
+            instance=version_one,
+            project=self.project,
+        )
+        assert not form.is_valid()
+        assert "slug" in form.errors
+
+    def test_cant_change_slug_machine_created_versions(self):
+        version = self.project.versions.get(slug="latest")
+        assert version.machine
+
+        form = VersionForm(
+            {
+                "slug": "change",
+                "active": True,
+                "privacy_level": PUBLIC,
+            },
+            instance=version,
+            project=self.project,
+        )
+        assert form.is_valid()
+        version.refresh_from_db()
+        assert version.slug == "latest"
```

Comment:
This action should fail explicitly instead of silently. It should raise a validation error saying it's not possible to change the slug of these versions.

Reply:
That's how forms work when a field is marked as disabled. Users won't be able to edit it from the UI.